components/ChatApp.tsx

```diff
@@ -352,4 +262,64 @@ export class ChatApp extends React.Component<
             </main>
         );
     }
+
+    public async componentDidMount() {
+        this.chatClient = await Chat.create(this.props.token);
+        this.chatClient.on("channelAdded", (channel: any) => {
+            this.setState((prevState, props) => ({
+                channels: [...prevState.channels, channel.uniqueName],
+            }));
+        });
+        if (this.props.candidateName !== undefined) {
+            // Si un candidate a été invité
+            const previousChannel = this.channel || undefined;
+            try {
+                // je tente de créer un canal
+                this.channel = await this.chatClient.createChannel({
+                    uniqueName:
+                        this.props.username + " - " + this.props.candidateName,
+                });
+            } catch (error) {
+                // si le canal existe je récupère ses informations afin de le rejoindre
+                if (error.code === channelAlreadyExist) {
+                    // mettre l'id de l'un et de l'autre avec virgule entre les deux
+                    this.channel = await this.chatClient.getChannelByUniqueName(
+                        this.props.username + " - " + this.props.candidateName,
+                    );
+                }
+            }
+            this.channel.add(this.props.username);
```

Comment:
this channel.add return a promise https://media.twiliocdn.com/sdk/js/chat/releases/2.1.0/docs/Channel.html#add__anchor

so:

`await this.channel.add // ...`